ozaria/site/components/interactive/PageInteractive/insertCode/index.vue

```diff
@@ -187,7 +188,24 @@
       }
     },
 
+    mounted () {
+      window.addEventListener('resize', this.onResize)
+      this.onResize()
+    },
+
+    beforeDestroy () {
+      window.removeEventListener('resize', this.onResize)
+    },
+
     methods: {
+      onResize () {
+        if (!this.codemirrorReady) {
+          return
+        }
+
+        window.requestAnimationFrame(() => this.codemirror.setSize('100%', '100%'))
```

Comment:
This seems like a way to defer the `setSize` method call. I'm interested in what the pros are of this approach.

Reply:
Yup, this queues the changes until right before the browsers next render.  It's a performance optimization (often used in animations) so allow visual updates to only run on browser repaints.  This is usually synced with the browser frame rate.